src/test/java/com/facebook/com/bitEncoderDecoder/bitwise/BitwiseTransmitterImplTest.java

```diff
@@ -0,0 +1,61 @@
+package com.facebook.com.bitEncoderDecoder.bitwise;
+
+import com.facebook.bitEncoderDecoder.app.Transmitter;
+import com.facebook.bitEncoderDecoder.bitwise.BitwiseTransmitterImpl;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class BitwiseTransmitterImplTest {
+
+    Transmitter transmitter = new BitwiseTransmitterImpl();
+
+    @DisplayName("Should send() outputs changed string")
+    @ParameterizedTest
+    @ValueSource(strings = {"abc", "test", "Ala ma kota"})
+    void send(String source){
+        assertNotEquals(source, transmitter.send(source));
+    }
+
+    @DisplayName("Should send() change only 1 bit in each byte")
+    @ParameterizedTest
+    @ValueSource(strings = {"abc", "test", "Ala ma kota"})
+    void sendAndCheckBites(String source){
+        // given
+        int[] onesInSource = countOnesInString(source);
+
+        // when
+        String target = transmitter.send(source);
+        int[] onesInTarget = countOnesInString(target);
+
+        // then
+        assertEquals(onesInSource.length, onesInTarget.length);
+        for (int i = 0; i < onesInSource.length; i++){
+            boolean oneBitIsTaken = onesInSource[i] - 1 == onesInTarget[i];
+            boolean oneBitIsAdded = onesInSource[i] + 1 == onesInTarget[i];
```

Comment:
Did you think about reverse name of this boolean (and similiar): oneBitIsTaken to TakenIsOneBit ? I think it could be usefull, on automatic sugesstion from Intellij (or eclipse).

Reply:
If you think it is easier to understand, I will change it.